### ISSUE TITLE
Adding laptops using AD

### ISSUE THREAD

**Issue by tashman89** [grokability/snipe-it#8319]
**Server (please complete the following information):**
 - Snipe-IT 9
 - OS: [Windows]
 - Web Server: [IIS]
 - PHP Version 7

**Is your feature request related to a problem? Please describe.**
Now I don't have a problem . 

**Describe the solution you'd like**
Dears support team,
We have about 600 laptops in our company and we need your kind advise about how we can adding them to the system  using active directory or LDAP . 

**Describe alternatives you've considered**

**Additional context**

Br,

**Comment by tashman89** [comment]
Dear

Can anyone help me please if there is any way to add 500 laptops one time using AD or LDAP.


Br,

**Comment by ajsy** [comment]
export the data from AD, and import the file into snipe-it

**Comment by tashman89** [comment]
Thanks Ajsy

**Comment by tashman89** [comment]
Dear Ajsy ,

could you please advise if you have the steps .

Thanks

**Comment by sjackson0109** [comment]
you can use many methods of extracting AD computer related data..
1) Microsoft Excel...
Ribbon> Data>GetData>From Other Sources>From Active Directory
Enter the FQDN of the domain, if not auto-populated > OK
Choose the Navigator (object Class) called "computer"
Use the Connections pane on the right, click EDIT and have a go at adding other columns... it's not simple, but it's really effective.

Here is an example of the 'advanced powerquery code':
let
    Source = ActiveDirectory.Domains("local/authenticated domain name goes here"),
    domain name = Source{[Domain="target/trusted domain/FQDN goes here"]}[#"Object Categories"],
    computers = domain name{[Category="computer"]}[Objects],
    #"Expanded user" = Table.ExpandRecordColumn(computers, "computer", {"msDS-User-Account-Control-Computed", "userAccountControl"}, {"user.msDS-User-Account-Control-Computed", "user.userAccountControl"}),
    #"Removed Other Columns" = Table.SelectColumns(#"Expanded user",{"user.userAccountControl", "computer", "displayName"})
in
    #"Removed Other Columns"

2) Microsoft Powershell...
Start>Run>Powershell.exe
Import-Module ActiveDirectory
Get-ADComputer -Filter * -Property * | Select-Object Name,OperatingSystem,OperatingSystemVersion,ipv4Address | Export-CSV ADcomputerslist.csv -NoTypeInformation -Encoding UTF8

3) Old-School Command line version of DSQuery:
Start>Run>
CMD
DSQUERY COMPUTER "OU=servers,DC=mydomain,DC=com" -o rdn -limit 1000 > c:\machines.txt

4) Windows GUI version of DSQuery:
Start>Run>
%SystemRoot%\SYSTEM32\rundll32.exe dsquery,OpenQueryWindow
Find menu > Computers
Click Find
 I don't think you can save via this method anymore, but you can do with ADUC (start>run>dsa.msc) but you need 'Remote Server Administration Tools' installed on your client.

**Comment by sjackson0109** [comment]
Yes,  importing assets autonomously from AD would be tremendously useful.

I’ve worked for several MSPs; and most of them advise clients to use computer naming conventions matching:

Hostname: <serial/tag>

I also have introduced dozens of auto-startup GPO power shell/VBA riots to collect local hardware data (make/model/Mac/audio device/battery status etc) and put the data within the AD computer object.. there are dozens of Pre-existing fields, although these can be easily expanded upon - suggest doing it in a lab first)..By sync’ing AD computers (objectclass=computer) for a given OU, it would save on the manual extract/cleanup in excel and re-import.

Also - AD computer objects have a ManagedBy attribute (similar to Manager for user objects).. where most MSPs assign the end user.  It’s a multi-valued array, but the ADUC gui doesn’t let you add more than a single value DN in that attribute.

**Comment by alex13p** [comment]
Will this ever get added?
